packages/react/src/utilities/hooks/useIsSmallScreen.ts

```diff
@@ -0,0 +1,20 @@
+import { useCallback, useEffect, useState } from "react";
+
+export const useIsSmallScreen = (breakPointToCheck: number) => {
+  const [isSmallScreen, setIsSmallScreen] = useState<boolean>(
+    window.innerWidth < breakPointToCheck,
+  );
+
+  const handler = useCallback(() => {
+    setIsSmallScreen(window.innerWidth < breakPointToCheck);
+  }, [breakPointToCheck]);
```

Comment:
ici l'appel au `setIsSmallScreen `va te générer de renderer superflu.
Voici une syntaxe plus optimiser
```suggestion
  const handler = useCallback(() => {
    const newState = window.innerWidth < breakPointToCheck;
    if (isSmallScreen !== newState) {
      setIsSmallScreen(newState);
    }
  }, [breakPointToCheck, isSmallScreen]);
```

D'ailleurs quitte à faire du react 18 autant même utiliser les nouvelles API et ce serait l'ocasion d'utiliser le useSyncExternalStore https://react.dev/reference/react/useSyncExternalStore#subscribing-to-a-browser-api

```typescript
const useIsSmallScreen = (breakPointToCheck: number) => {
  const subscribe = useCallback((listener: () => void) => {
    window.addEventListener('resize', listener);
    return () => {
      window.removeEventListener('resize', listener);
    };
  }, []);

  const getSnapshot = useCallback(() => {
    return window.innerWidth < breakPointToCheck;
  }, [breakPointToCheck]);

  return useSyncExternalStore(subscribe, getSnapshot);
};
```

Reply:
je connaissais pas, top ! merci